codebasin/config.py

```diff
@@ -444,19 +442,19 @@ def load_database(dbpath, rootdir):
         # - relative to a directory
         # - as an absolute path
         filedir = rootdir
-        if "directory" in e:
-            if os.path.isabs(e["directory"]):
-                filedir = e["directory"]
+        if command.directory is not None:
+            if os.path.isabs(command.directory):
+                filedir = command.directory
             else:
                 filedir = os.path.realpath(
                     rootdir,
-                    os.path.join(e["directory"]),
+                    os.path.join(command.directory),
                 )
 
-        if os.path.isabs(e["file"]):
-            path = os.path.realpath(e["file"])
+        if os.path.isabs(command.filename):
+            path = os.path.realpath(command.filename)
```

Comment:
I know I've brought this up in previous reviews, but bringing it up again this time because `codebasin/source.py` uses `pathlib.Path`. Is it used in this file for consistency's sake?

Reply:
I'm just trying to keep the change sets small, so that it's easier to identify where things went wrong if they start breaking.  I've started trying to use `pathlib.Path` whenever I write new code, but I'm not going back to change any existing code.

I only changed these lines because it seemed dangerous to use both representations of the compile command being used within the same loop -- it would be very easy to update one and forget to change the other.